api/routes/utils.js

```diff
@@ -545,29 +544,29 @@ var utils = {
     return expert.categories[i].reps;
   },
 
-  // Update an investor making an investment for a given category
-  // Returns null if success, error message otherwise
-  updateTransactionFromUser: function(fromUser, toUser, category, amount, toUserReps, investmentId) {
-    var portfolio = fromUser.portfolio;
+  getTransactionPortfolioIndex: function(amount, fromUser, toUser, toUserReps, investmentId, category) {
 
     // Find the portfolio entry that should be updated
-    var index = -1;
-    var length = portfolio.length;
-    for (var i = 0; i < length; i++) {
-      if (portfolio[i].category === category) {
-        var investments = portfolio[i].investments;
-        index = i;
-        break;
-      }
+    var index = this.getPortfolioIndex(fromUser, category.name);
+    if (index !== -1) {
+      return index;
     }
 
-    // The from user is not an investor for this category (ERROR!)
-    if (index === -1) {
-      return 'User is not an investor for this category';
-    }
+    // If the user is not an investor for this category, add it
+    fromUser.portfolio.push({ category: category.name, id: category._id, percentile: 0, investments: [] });
```

Comment:
Does `push` only add the category if it does not yet exist in the user's portfolio (Is it idempotent?)?